### ISSUE TITLE
Non draining NEAR recovery \ login link

### ISSUE THREAD

**Issue by intmainreturn00** [near/near-wallet#2494]
## Story

As a developer, I want generate & pre-fund implicit accounts for users and send them a link they can use to log-in into their account with near wallet. I found this recovery deep-link (for example): 

```
https://wallet.testnet.near.org/recover-with-link/<accountId>/<seed-phrase>
```

And it works great. But there is one problem with such link - even if user already have keys for account in their browser storage (meaning they already "recovered" their account via link once) the link will still drain some near and make new access key. 

Is it possible to have a link which first checks the keys availability and only if they are not found, do the recover. Something like recover \ login link?

## Acceptance Criteria

- [ ] User can recover their account by going through link the first time
- [ ] User don't spend additional near going through the same link over and over again if browser storage has needed keys

**Comment by agileurbanite** [comment]
Investigating

**Comment by andy-haynes** [comment]
Hi @intmainreturn00, thank you for the thorough description. I believe I have a fix that we can have on testnet for validation by the end of the week.

**Comment by andy-haynes** [comment]
Unfortunately there are some edge cases supported by the current `recover-with-link` route that need to be considered in addressing this issue. Currently we are considering making this functionality opt-in, i.e. users whose browsers are logged in to the account may choose to continue on with recovery if either:
- the user wishes to recover the implicit account associated with the named account in the URL
- the user wishes to recover a corrupted account that appears to be logged in but is in reality non-functional due to corrupted localStorage

There is active discussion on the PR and we hope to have a solution ready once we have decided on how best to present the user with this option.

**Comment by intmainreturn00** [comment]
@agileurbanite thanks for alternatives, I did consider this, but it didn't seems suitable (correct me if I'm wrong):
* for our project we want to use **implicit accounts**, mostly for privacy concerns specific to our project. linkdrops are great but they works with named accounts. 
* also with the current approach, we can login \ onboard user into both near wallet and our *in-development* token-wallet by just one click, so that crypto curious users can avoid dealing with seed phrases themselves (and at the same time - we don't keep seeds and so crypto natives can keep it in a way they find suitable).

So it seems like there is still some utility in having non-draining recovery link. Hope that edge cases could be resolved.
🙌

**Comment by agileurbanite** [comment]
@Patrick1904 @MaximusHaximus is this issue now closed by #2521 ?

**Comment by Patrick1904** [comment]
> @Patrick1904 @MaximusHaximus is this issue now closed by #2521 ?

Yes this shipped to mainnet with https://github.com/near/near-wallet/pull/2570

**Comment by intmainreturn00** [comment]
Thanks, it works with https://wallet.mainnet.near.org/recover-with-link/<accountId>/<seed> 🙏
